### PR TITLE
Run CI on Composer v2

### PR DESCRIPTION
**Pull request by szepeviktor**

### CODE REVIEW

**Review by iruzevic — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,10 +41,10 @@ jobs:
                   ${{ runner.os }}-composer-
 
             - name: Install composer packages
-              run: composer install --no-progress
+              run: composer update --no-progress
```
Comment (dingo-d) [inline .github/workflows/ci.yml:44]: Won't this update the packages?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -41,10 +41,10 @@ jobs:
...
 
             - name: Check coding standards using PHPCS
-              run: composer standards:check -- --runtime-set testVersion ${{ matrix.php }}-
+              run: composer run-script standards:check -- --runtime-set testVersion ${{ matrix.php }}-
```
Comment (dingo-d) [inline .github/workflows/ci.yml:47]: The script runs without the `run-script` addition. Is there any benefit to adding it?

**Review by dingo-d — COMMENTED** [review]
I've left some comments.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (szepeviktor) [reply .github/workflows/ci.yml:47]: @dingo-d As we do not have a lock file, there is no way to have fixed versions.
BTW Having a lock file a really bad idea: e.g. testing on different PHP version etc.
https://github.com/ergebnis/php-library-template/blob/main/.github/workflows/integrate.yaml#L28-L29
Testing could be done with lowest and normal (highest) dependencies.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (szepeviktor) [reply .github/workflows/ci.yml:44]: @dingo-d As we do not have a lock file, there is no way to have fixed versions.
BTW Having a lock file a really bad idea: e.g. testing on different PHP version etc.
https://github.com/ergebnis/php-library-template/blob/main/.github/workflows/integrate.yaml#L28-L29
Testing could be done with lowest and normal (highest) dependencies.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (szepeviktor) [reply .github/workflows/ci.yml:47]: Writing explicit code aims review.
I am not the one to convince you :)

Comment (szepeviktor) [reply .github/workflows/ci.yml:47]: e.g. do you understand `grep -rFnwh`? :-D

Comment (dingo-d) [reply .github/workflows/ci.yml:47]: Yeah, but running `composer` will show extra scripts 😄  I mean, a bit more verbosity doesn't hurt 😄

**Review by dingo-d — APPROVED** [review]

**Pull request merged**